### ISSUE TITLE
KeyError: u'FLASK_AUTH_REALM'

### ISSUE THREAD

**Issue by bth5032** [carsongee/flask-htpasswd#16]
Can't get this to work at all, when I visit a page decorated with @htpasswd.required, I get a crash with the traceback shown in the image: http://imgur.com/IapkLWm. 

I followed your basic example to the letter and even included the line 

`app.config['FLASK_AUTH_REALM'] = 'admin_panel'`

where admin_panel is the realm for authentication as far as I understand. I tried several other values for Realm including the default you suggest 'Login Required'. I could not get basic functionality working.

**Comment by carsongee** [comment]
That is strange, I just tried with this code, and it worked:

``` python
import flask
from flask.ext.htpasswd import HtPasswdAuth

app = flask.Flask(__name__)
app.config['FLASK_HTPASSWD_PATH'] = '.htpasswd'
app.config['FLASK_SECRET'] = 'Hey Hey Kids, secure me!'
app.config['FLASK_AUTH_REALM'] = 'Hey you'
htpasswd = HtPasswdAuth(app)


@app.route('/')
@htpasswd.required
def index(user):
    return 'Hello {user}'.format(user=user)

app.run(debug=True)
```

Does it work without that defined, or is it possible the config is set after you initialize the `HtPasswdAuth` object?

**Comment by bth5032** [comment]
Ahh okay, I played around with this a bit more after reading your message. This wasn't a bug, the issue was that I wasn't accepting "user" in my function. It would be nice if that threw some sort of proper error though. Thanks for the help.

EDIT: I'm trying to impliment this on: https://github.com/bth5032/fstate.  What worked was making another minimal server that followed your example, logging in with that, then starting my original server again I stayed logged in. But if I open a new browser and try to access my restricted pages, I'm getting the same error. So I should not have closed this issue.
